### DISCUSSION TITLE
Opti Specifying discrete option

### DISCUSSION THREAD

**Discussion by rtapia8484** [casadi/casadi#3950] [Q&A] [answered]
Hello,

from reading https://github.com/casadi/casadi/wiki/FAQ:-how-to-work-with-discrete-mixed-integer-problems%3F
I understand I have to specify a cell array stating which variables are discrete,

is the example below correct (nx=6,nu=1, nobs*2=6) in general 

discrete = []

for k in range(N):
    Z.append(opti.variable(nx))
    discrete.append([False, False, False, False, False, False])
	# E.append(opti.variable(nx))
    I.append(opti.variable(nu))
    discrete.append([False])
    BV.append(opti.variable(nobs*2))
    discrete.append([True, True, True, True, True, True])
    L.append(opti.variable())
    discrete.appened([False])
    
Z.append(opti.variable(nx))
discrete.append([False, False, False, False, False, False])
L.append(opti.variable())
discrete.append([False])

Best

**Comment by jgillis** [comment, marked as answer]
As of 3.6.6, there is an easier way to do this in Opti. I adapted the wiki page accordingly.

**Reply by rtapia8484** [reply]
thanks!
